### PR TITLE
This closes #83 adding commit info in the version of Apia

### LINKED ISSUE #83: Include Git hash in Apia version
The `apia --version` option could include the git commit at building time, i.e. the hash of the git top commit.

This feature was requested in Agda in https://github.com/agda/agda/issues/2023 and it was added in https://github.com/agda/agda/commit/5a1361f67be69aeff870a60a5f4152b4d2477ee0.

I would like the same feature for Apia but *without* using Template Haskell.

@jonaprieto, I assigned you this issue.

### PR DESCRIPTION
**Pull request by jonaprieto**
This PR adds information in the version of Apia like this:

```
$ apia --version
Apia version 1.0.1-4326348
```

A final comment, I am using a deprecated constructor `versionTags` (See GHC ticket #2496) but it doesn't seem about our concern at least for now. 

References:
- https://gist.github.com/hvr/656738
- [vcs-revision](https://github.com/jkff/vcs-revision/blob/master/Distribution/VcsRevision/Git.hs)
- [gitrev](https://github.com/acfoltzer/gitrev/blob/master/src/Development/GitRev.hs)

### CODE REVIEW

**Review by asr — CHANGES_REQUESTED** [review]
Good job!

Please fix the test-suite and the HLint warnings in `Setup.hs`.

**Comment by asr** [comment]
The test-suite continues failing...

File: Setup.hs [hunk 1]
```diff
@@ -1,30 +1,54 @@
 
 -- | Setup.hs module
 
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
```
Comment (asr) [inline Setup.hs]: According to the GHC documentation (various versions) this flag is called `warnings-deprecations` so you should use `-fno-warn-warnings-deprecations`. In fact, the synonymous name `deprecations` is not documented, and GHC shouldn't report something like

```
dist/setup/setup.hs:56:37: warning: [-Wdeprecations]
 ```

but

```
dist/setup/setup.hs:56:37: warning: [-Wwarnings-deprecations]
 ```

Please create a MVE a file an GHC issue.

Comment (asr) [reply Setup.hs]: > Please create a MVE a file an GHC issue.

CC'ing to me. My GHC Trac user is asr.

**Comment by asr** [comment]
The test-suite continues failing...

File: Setup.hs [hunk 2]
```diff
@@ -1,30 +1,54 @@
...
+{-# LANGUAGE ScopedTypeVariables       #-}
+
```
Comment (asr) [inline Setup.hs:7]: Please add `-Wall` and fix the warnings.

**Review by asr — CHANGES_REQUESTED** [review]

**Review by asr — CHANGES_REQUESTED** [review]
The test-suite continues failing...

**Comment by jonaprieto** [comment]
Where can I find `shelltest` binaries? I couldn't find it to install in OSX.

**Review by asr — APPROVED** [review]

**Comment by asr** [comment]
@jonaprieto, I squashed the commits and added some information from the OP.

Please delete the branch.

Thanks!

**Pull request merged**